pennylane/ops/op_math/prod.py

```diff
@@ -324,6 +324,10 @@ def sparse_matrix(self, wire_order=None):
         full_mat = reduce(sparse_kron, mats)
         return math.expand_matrix(full_mat, self.wires, wire_order=wire_order)
 
+    @property
+    def has_sparse_matrix(self):
+        return self.pauli_rep is not None or all(op.has_sparse_matrix for op in self)
+
```

Comment:
Could you add the same property for `Sum` as well?

Reply:
Done